File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.rs
impl Solution {
    pub fn two_sum(numbers: Vec<i32>, target: i32) -> Vec<i32> {
        let mut i = 0;
        let mut j = numbers.len()-1;

        while i < j {
            if numbers[i]+numbers[j] < target {
                i+=1;
            } else if numbers[i]+numbers[j] > target {
                j-=1;
            } else {
                return Vec::from([i as i32 +1, j as i32 +1]);
            }
        }

        return Vec::new();
    }
}

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        freq = {}

        for i in range(len(numbers)):
            if target - numbers[i] in freq:
                return [freq[target-numbers[i]]+1, i+1]
            else:
                freq[numbers[i]] = i
        
        return [0, 0]